internal/jptrace/spanhash.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package jptrace
+
+import (
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"unsafe"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type SpanHasher struct {
+	marshaler ptrace.Marshaler
+}
+
+func NewSpanHasher() *SpanHasher {
+	protoMarshaler := &ptrace.ProtoMarshaler{}
+	return &SpanHasher{
+		marshaler: protoMarshaler,
+	}
+}
+
+func (s *SpanHasher) SpanHash(trace ptrace.Traces) (int64, error) {
```

Comment:
`ptrace.Traces` is a collection of spans, it doesn't make sense to ask for a single hash. It should be taking `ptrace.Span` as input.